lua/telescope/_extensions/project.lua

```diff
@@ -0,0 +1,62 @@
+local has_telescope, telescope = pcall(require, 'telescope')
+
+if not has_telescope then
+	error('This plugins requires nvim-telescope/telescope.nvim')
+end
+
+local actions = require("telescope.actions")
+local finders = require("telescope.finders")
+local pickers = require("telescope.pickers")
+local conf = require("telescope.config").values
+local builtin = require("telescope.builtin")
+local project_dirs_file = vim.fn.stdpath('data') .. '/telescope-projects.lua'
+
+local select_project = function(opts, projects, run_task_on_selected_project)
+	pickers.new(opts, {
+		prompt_title = 'Select a project',
+		results_title = 'Projects',
+		finder = finders.new_table {results = projects},
+		sorter = conf.file_sorter(opts),
+		attach_mappings = function(prompt_bufnr)
+			local on_project_selected = function()
+				local selection = actions.get_selected_entry(prompt_bufnr)
+				actions.close(prompt_bufnr)
+				run_task_on_selected_project(selection.value)
+			end
+			actions.goto_file_selection_edit:replace(on_project_selected)
+			return true
+		end
+	}):find()
+end
+
+local project = function(opts)
+	opts = opts or {}
+
+	-- TODO figure this out:
+
+	-- local project_dirs = {}
+	-- for line in io.lines(project_dirs_file) do
+	-- 	table.insert(project_dirs, line)
+	-- end
+
+	local project_dirs = {["nvim config"] = "~/.config/nvim"}
+	local projects = {}
+
+	local search_selected_project = function(selection)
+		local selected_directory = project_dirs[selection]
+		builtin.find_files({cwd = selected_directory})
+	end
+
+	-- TODO allow a way to choose between the above function and this one.
+	local open_path_at_selected_project = function(selection)
+		vim.api.nvim_command('Explore ' .. project_dirs[selection])
+	end
+
+	for k in pairs(project_dirs) do
+		table.insert(projects, k)
+	end
```

Comment:
It should be `for k, v in pairs()`.
Also we most probably wanna display the name and want the path as value. So you could add something like this.
```lua
table.insert(projects, {
  value = v, -- would be path,
  display = k, -- would be name,
  ordinal = k,
})
```

I think the default entry maker will get this one right. If not and you end up with no visible items. Feel free to ask and i help you write a new one.

Reply:
Ah I noticed I wasn't using v previously, so I removed it, but this sounds nice

Reply:
@Conni2461 this one isn't working for me, using this with `finders.new_table { results = projects }`

Reply:
If I do `selection.value.value` I can get the path back, and `selection.value.display` brings back the display, but it's not working to just use the object as is in finder.new_table

Reply:
I've noticed that it's an array with the project data inside it now, like: 
`{ { display = 'nvim config', ordinal = 'nvim config', value = '~/.config/nvim } }`
so I tried passing in projects[1] also to the select_project function but with no luck. finders.new_table doesn't seem to be picking up the display property

Reply:
You probably need to write an entry_maker. Take a look at `lua/telescope/builtin/internal.lua` at symbols for example. Line 89 - 136